frontend/kubernetes/plugin.go

```diff
@@ -244,6 +258,24 @@ func newKubernetesPlugin(
 		},
 	)
 
+	// Setting up a watch for VolumeSnapshot CR
+	ret.volumeSnapshotSource = &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			return kubeSnapClient.VolumesnapshotV1alpha1().VolumeSnapshots(v1.NamespaceAll).List(options)
+		},
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			return kubeSnapClient.VolumesnapshotV1alpha1().VolumeSnapshots(v1.NamespaceAll).Watch(options)
+		},
+	}
+	_, ret.volumeSnapshotController = cache.NewInformer(
+		ret.volumeSnapshotSource,
+		&k8ssnapshotv1alpha1.VolumeSnapshot{},
+		KubernetesSyncPeriod,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: ret.addVolumeSnapshot,
```

Comment:
This should also handle update correctly - Might as well also set this callback for the handler.